src/KafkaSchemaRegistryApiApiClient.php

```diff
@@ -0,0 +1,212 @@
+     */
+    public function getSubjectCompatibilityLevel(string $subjectName): ?string
+    {
+        try {
+            $results = $this->registryClient->call('GET', sprintf('/config/%s', $subjectName));
+            return $results['compatibilityLevel'];
+        } catch (ResourceNotFoundException $e) {
+            return $this->getDefaultCompatibilityLevel();
+        }
+    }
+
+    /**
+     * @param string $subjectName
+     * @param string $level
+     * @return bool
+     */
+    public function setSubjectCompatibilityLevel(string $subjectName, string $level = self::LEVEL_FULL): bool {
+
+        $this->registryClient->call(
+            'PUT',
+            sprintf('config/%s', $subjectName),
+            ['compatibility' => $level]
+        );
+
+        return true;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDefaultCompatibilityLevel(): string {
+
+        $results = $this->registryClient->call('GET', 'config');
+
+        return $results['compatibilityLevel'];
+    }
+
+    /**
+     * @param string $level
+     * @return bool
+     */
+    public function setDefaultCompatibilityLevel(string $level = self::LEVEL_FULL): bool {
+        $this->registryClient->call('PUT', 'config', ['compatibility' => $level]);
+        return true;
+    }
+
+    /**
+     * @param string $subjectName
+     * @param string $schema
+     * @return string|null
+     */
+    public function getVersionForSchema(string $subjectName, string $schema): ?string
+    {
+        try {
+            $results = $this
+                    ->registryClient
+                    ->call(
+                        'POST',
+                        sprintf('/subjects/%s/', $subjectName),
+                        $this->prepareSchemaData($schema)
+                    ) ?? [];
+
+            return (string) $results['version'];
+        } catch (ResourceNotFoundException $e) {
+            return null;
+        }
+    }
+
+    /**
+     * @param string $subjectName
+     * @return array
+     */
+    public function deleteSubject(string $subjectName): array
+    {
+        return $this
+            ->registryClient
+            ->call(
+                'DELETE',
+                sprintf('/subjects/%s/', $subjectName)
+            ) ?? [];
+    }
+
+    /**
+     * @param string $subjectName
+     * @return string|null
+     * @throws RequestException
+     */
+    public function getLatestSubjectVersion(string $subjectName): ?string
+    {
+        try {
+            $schemaVersions = $this->getAllSubjectVersions($subjectName);
+            $lastKey = array_key_last($schemaVersions);
+            return $schemaVersions[$lastKey];
+        } catch (ResourceNotFoundException $e) {
+            return null;
+        }
+    }
+
+    /**
+     * @param string $schema
+     * @return array
+     */
+    private function prepareSchemaData(string $schema): array
+    {
+        return ['schema' => json_encode(json_decode($schema, true))];
```

Comment:
why do you decode then encode again?

Reply:
I followed what they did in flix :D But shortened a bit.
I am not sure of a reason.

I will change then this....

Reply:
On slack discussion, we agreed that this is because of formatting